patch-package
--- a/node_modules/react-scripts-ts/config/webpack.config.dev.js
+++ b/node_modules/react-scripts-ts/config/webpack.config.dev.js
@@ -126,6 +126,7 @@ module.exports = {
       new ModuleScopePlugin(paths.appSrc, [paths.appPackageJson]),
       new TsconfigPathsPlugin({ configFile: paths.appTsConfig }),
     ],
+    symlinks: false,
   },
   module: {
     strictExportPresence: true,
