patch-package
--- a/node_modules/typescript/lib/tsc.js
+++ b/node_modules/typescript/lib/tsc.js
@@ -24374,7 +24374,9 @@ var ts;
                             undefined;
                 }
                 if (t.flags & 524288) {
-                    return stringType;
+                    const baseType = getBaseConstraint(t.type);
+                    const baseIndex = baseType ? getIndexType(baseType) : undefined;
+                    return baseIndex && baseIndex !== unknownType ? getBaseConstraint(baseIndex) : undefined;
                 }
                 if (t.flags & 1048576) {
                     var transformed = getSimplifiedIndexedAccessType(t);
--- a/node_modules/typescript/lib/tsserver.js
+++ b/node_modules/typescript/lib/tsserver.js
@@ -25768,7 +25768,9 @@ var ts;
                             undefined;
                 }
                 if (t.flags & 524288) {
-                    return stringType;
+                    const baseType = getBaseConstraint(t.type);
+                    const baseIndex = baseType ? getIndexType(baseType) : undefined;
+                    return baseIndex && baseIndex !== unknownType ? getBaseConstraint(baseIndex) : undefined;
                 }
                 if (t.flags & 1048576) {
                     var transformed = getSimplifiedIndexedAccessType(t);
--- a/node_modules/typescript/lib/tsserverlibrary.js
+++ b/node_modules/typescript/lib/tsserverlibrary.js
@@ -32040,7 +32040,9 @@ var ts;
                             undefined;
                 }
                 if (t.flags & 524288 /* Index */) {
-                    return stringType;
+                    const baseType = getBaseConstraint(t.type);
+                    const baseIndex = baseType ? getIndexType(baseType) : undefined;
+                    return baseIndex && baseIndex !== unknownType ? getBaseConstraint(baseIndex) : undefined;
                 }
                 if (t.flags & 1048576 /* IndexedAccess */) {
                     var transformed = getSimplifiedIndexedAccessType(t);
--- a/node_modules/typescript/lib/typescript.js
+++ b/node_modules/typescript/lib/typescript.js
@@ -29995,7 +29995,10 @@ var ts;
                             undefined;
                 }
                 if (t.flags & 524288 /* Index */) {
-                    return stringType;
+                    const baseType = getBaseConstraint(t.type);
+                    const baseIndex = baseType ? getIndexType(baseType) : undefined;
+                    return baseIndex && baseIndex !== unknownType ? getBaseConstraint(baseIndex) : undefined;
+
                 }
                 if (t.flags & 1048576 /* IndexedAccess */) {
                     var transformed = getSimplifiedIndexedAccessType(t);
--- a/node_modules/typescript/lib/typescriptServices.js
+++ b/node_modules/typescript/lib/typescriptServices.js
@@ -29995,7 +29995,9 @@ var ts;
                             undefined;
                 }
                 if (t.flags & 524288 /* Index */) {
-                    return stringType;
+                    const baseType = getBaseConstraint(t.type);
+                    const baseIndex = baseType ? getIndexType(baseType) : undefined;
+                    return baseIndex && baseIndex !== unknownType ? getBaseConstraint(baseIndex) : undefined;
                 }
                 if (t.flags & 1048576 /* IndexedAccess */) {
                     var transformed = getSimplifiedIndexedAccessType(t);
